### ISSUE TITLE
adding (?) typical geometric helpers in cwapi?

### ISSUE THREAD

**Issue by fbpyr** [cwapi3d/cwapi3dpython#42]
After a brief discussion with @Brunner246 I would like to open exchange of ideas on the following topic: 

### Adding typical geometric helpers to cwAPI?

While it is true, that most of them could be implemented by consumers of cwAPI, often they might:
* bring performance benefits when implemented in C++ inside cadwork (some potentially already implemented)
* make it easier for newcomers to the api to have similar helpers as in other 3d-app APIs (rh, rvt, blender, freecad, etc)
* would be a canonical goto-tool for less reinventing the wheel

### non-exaustive example list of typical geometric helpers:
* **points**
  * closest point on line
  * closest point on plane
  * point z-positive/z-negative (beyond/above) to plane
  * point in global bounding box
  * point in local bounding box
  * point is in solid (already implemented to my knowledge)

* **lines**
  * project line onto plane
  * lines are on same plane
  * lines form closed loop

* **solids**
  * solid is in contact with other solid (already implemented to my knowledge)
  * solid is in collision with other solid (already implemented to my knowledge)

**Comment by Brunner246** [comment]
Hi @fbpyr 

Thanks for the inputs.

We started implementing some functions like point on line, point on solid, solids in contact, etc..

Furthermore, we created a repo called cwmath which can be used to implement mathematical/geometrical functions. 
The idea of this project is to share functionalities between the cadwork API users. 

https://github.com/cwapi3d/cwmath
https://docs.cadwork.com/projects/cwmath/en/latest/
